certbot/compat.py

```diff
@@ -117,56 +117,9 @@ def readline_with_timeout(timeout, prompt):
         # So no timeout on Windows for now.
         return sys.stdin.readline()
 
-def lock_file(fd):
-    """
-    Lock the file linked to the specified file descriptor.
-
-    :param int fd: The file descriptor of the file to lock.
-
-    """
-    if 'fcntl' in sys.modules:
-        # Linux specific
-        fcntl.lockf(fd, fcntl.LOCK_EX | fcntl.LOCK_NB)
-    else:
-        # Windows specific
-        msvcrt.locking(fd, msvcrt.LK_NBLCK, 1)
-
-def release_locked_file(fd, path):
-    """
-    Remove, close, and release a lock file specified by its file descriptor and its path.
-
-    :param int fd: The file descriptor of the lock file.
-    :param str path: The path of the lock file.
-
-    """
-    # Linux specific
-    #
-    # It is important the lock file is removed before it's released,
-    # otherwise:
-    #
-    # process A: open lock file
-    # process B: release lock file
-    # process A: lock file
-    # process A: check device and inode
-    # process B: delete file
-    # process C: open and lock a different file at the same path
-    try:
-        os.remove(path)
-    except OSError as err:
-        if err.errno == errno.EACCES:
-            # Windows specific
-            # We will not be able to remove a file before closing it.
-            # To avoid race conditions described for Linux, we will not delete the lockfile,
-            # just close it to be reused on the next Certbot call.
-            pass
-        else:
-            raise
-    finally:
-        os.close(fd)
-
 def compare_file_modes(mode1, mode2):
     """Return true if the two modes can be considered as equals for this platform"""
-    if 'fcntl' in sys.modules:
+    if fcntl:
```

Comment:
I take an opinionated approach here, influenced by the pythonic credo "easier to ask for forgiveness than permission" and good practices in the javascript world: instead of imperatively rely on the platform nature, rely on a capability (here fcntl is present), and do it by "try then handle exception".

You gave in another comment the elegant way of making a try/except/else, that fits well in this philosophy, I will do try that.

Reply:
Nothing here indeed. This code will soon disappear anyway. Until that, I will rely on os.name. Or sys.platform, I do not now what is more canonical.

Reply:
Seems os.name to be more appropriate to check POSIX compliant vs Windows, and this what is needed here.